interfaces/dbus/backend.go

```diff
@@ -179,6 +288,54 @@ func addContent(securityTag string, snippet string, content map[string]osutil.Fi
 	}
 }
 
+// snapNameFromServiceFile returns the snap name for the D-Bus service activation file.
+func snapNameFromServiceFile(filename string) (owner string, err error) {
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return "", err
+	}
+	snapKey := []byte("\nX-Snap=")
```

Comment:
Would this be tricked by `# not really X-Snap=foo`? Do we need a Ini-file parser?

Reply:
Note the `\n` at the start of the pattern. It can only match the beginning of a line.  As this is an ini file format where the key must be preceded by a group header, it can't be on the first line either.